### PR TITLE
#474 Add setters to the snackbar options, #477 Snackbar default settings

### PR DESCRIPTION
**Pull request by VladislavAntonyuk**
### Description of Change ###

Add setters to the snackbar options
Add Snackbar default settings

### Bugs Fixed ###
#474
#477 

### API Changes ###

Added: 
 
- static default properties for all options

Changed:

 - `ActionOptions` => `ToastOptions`
 - removed ctors for SnackBarOptions and ToastOptions. they were rather complex and required all options.

### PR Checklist ###

- [x] Has tests (if omitted, state reason in description)
- [x] Has samples (if omitted, state reason in description)
- [x] Rebased on top of main at time of PR
- [x] Changes adhere to coding standard

- [x] Updated [documentation](https://github.com/MicrosoftDocs/xamarin-communitytoolkit)

### CODE REVIEW

**Comment by VladislavAntonyuk** [comment]
@jfversluis, @AndreiMisiukevich could you please review? it has some breaking changes, it would be nice to fix them before release of stable version.

**Review by AndreiMisiukevich — DISMISSED** [review]
Looks good
Only one question. Is it important to rename "ActionOptions => ToastOptions"?
If others are OK with such renaming, I am OK too

**Comment by VladislavAntonyuk** [comment]
> Looks good
> Only one question. Is it important to rename "ActionOptions => ToastOptions"?
> If others are OK with such renaming, I am OK too

We use it with Toast. so ToastOptions is what user expect to use. ActionOptions is something really abstract. What is action for you? I think we should not associated it with Toast, SnackBar etc.

**Comment by pictos** [comment]
I will review this until tomorrow.

File: XamarinCommunityToolkit/Views/SnackBar/Options/ToastOptions.shared.cs [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+		/// Is Right to left
+		/// </summary>
+		public bool IsRtl { get; set; } = DefaultIsRtl;
+
+		public static bool DefaultIsRtl { get; set; } = false;
+
+		/// <summary>
+		/// The duration for the SnackBar.
+		/// </summary>
+		public int Duration { get; set; } = DefaultDuration;
```
Comment (pictos) [inline XamarinCommunityToolkit/Views/SnackBar/Options/ToastOptions.shared.cs:37]: Can we add the unit here, in the name of the property, or/and in the `summary` part?

File: XamarinCommunityToolkit/Views/SnackBar/Options/ToastOptions.shared.cs [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+
+		public static int DefaultDuration { get; set; } = 3000;
```
Comment (pictos) [inline XamarinCommunityToolkit/Views/SnackBar/Options/ToastOptions.shared.cs:39]: Can we add the unit here? (ms, us, seconds)

**Review by pictos — CHANGES_REQUESTED** [review]
@VladislavAntonyuk nice work, as always! I have just some considerations about names, to make it more clear to our end users. Please let me know what do you think.

**Comment by VladislavAntonyuk** [comment]
> @VladislavAntonyuk nice work, as always! I have just some considerations about names, to make it more clear to our end users. Please let me know what do you think.

Thank you! I found that XF has Duration property of MediaElement. To be consistent with it I think we can leave it Duration. But what do you think if I change the type to TimeSpan?

**Comment by pictos** [comment]
@VladislavAntonyuk I like the idea of TimeSpan.

**Review by pictos — APPROVED** [review]

**Review by AndreiMisiukevich — APPROVED** [review]

**Comment by Tommigun1980** [comment]
Thank you so much once again @VladislavAntonyuk! Looks fantastic.

**Pull request merged**